llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9439,6 +9418,20 @@ VPlanPtr LoopVectorizationPlanner::buildVPlan(VFRange &Range) {
   bool HasNUW = true;
   addCanonicalIVRecipes(*Plan, Legal->getWidestInductionType(), HasNUW,
                         DebugLoc());
+
```

Comment:
Would be good to reduce the disparity between native and legacy VPlan constructions (ultimately converging). Perhaps have the former also record Ingredient2Recipe for widened inductions it creates during VPInstructionsToVPRecipes(), to then use getRecipe(P) as above, rather than searching linearly for each as below?

Reply:
Yep, adjusted, thanks!